### ISSUE TITLE
Infix operator lint metadata is not helpful for object_usage_linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1497]
```r
lintr::lint("a <- function() {\n  1:10 %>% sum()\n}", lintr::object_usage_linter())
# <text>:1:6: warning: [object_usage_linter] no visible global function definition for ‘%>%’. Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
# a <- function() {
#      ^~~~~~~~~~~~
```

The linter anchors to the function definition, not to the usage of `%>%` itself. Similar goes for other infixes, e.g.

```r
lintr::lint("a <- function() {\n  1:10 %>% sum()\n}", lintr::object_usage_linter())
# <text>:1:6: warning: [object_usage_linter] no visible global function definition for ‘:=’. Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
# a <- function(x) {
#      ^~~~~~~~~~~~~
```

I guess this is `codetools`' fault, but is there something we can do to patch it up? Or maybe we have a bad XPath?

**Comment by MichaelChirico** [comment]
Native pipe also affected by this:

```r
lintr::lint("foo <- function(x) {
  x |>
    dplyr::mutate(day = as.POSIXct(round(timestamp, units = 'days')))
}", lintr::object_usage_linter())
# <text>:2:3: warning: [object_usage_linter] unused argument (units = "days").
#   x |>
#   ^~~~
```

**Comment by MichaelChirico** [comment]
Ah, nevermind, I think this is not the same issue.

```r
lintr::lint("foo <- function(x) {
  x |>
    sum(y)
}", lintr::object_usage_linter())
# <text>:3:9: warning: [object_usage_linter] no visible binding for global variable 'y'.
#     sum(y)
#         ^
```

correct answer here.
